atd-vze/src/views/Crashes/CrashChange.js

```diff
@@ -0,0 +1,1238 @@
+    UPSERT_MUTATION_DUMMY
+  );
+
+  const { loading, user } = useAuth0();
+
+  /**
+   * Mutations
+   */
+  const [deleteFromQueue] = useMutation(
+    gql`
+      ${RECORD_DELETE_CHANGE_RECORDS}
+    `
+  );
+
+  const [upsertRecordUpdates] = useMutation(
+    gql`
+      ${upsertRecordQuery}
+    `
+  );
+
+  const {
+    data: data,
+    error: error,
+    loading: loadingData,
+    refetch: refetch,
+  } = useQuery(GET_CRASH_CHANGE, {
+    variables: { crashId },
+  });
+
+  const {
+    data: secondaryData,
+    error: secondaryError,
+    loading: loadingSecondaryData,
+  } = useQuery(GET_CRASH_SECONDARY_RECORDS, {
+    variables: { crashId },
+  });
+
+  /**
+   * Wait for data to be retrieved, then change state.
+   */
+  useEffect(() => setRecordData(data), [data]);
+
+  /**
+   * Returns true if fieldName exists within the selectedFields array.
+   * @param {string} fieldName - The name of the field
+   * @returns {boolean}
+   */
+  const isFieldEnabled = fieldName => {
+    return selectedFields.includes(fieldName);
+  };
+
+  /**
+   * Sleeps a for a few milliseconds
+   * @param {int} milliseconds - The length of sleep in milliseconds
+   * @returns {Promise}
+   */
+  const sleep = milliseconds => {
+    return new Promise(resolve => setTimeout(resolve, milliseconds));
+  };
+
+  /**
+   * Returns a gql object ready to be executed.
+   * @param  {string} template - The GraphQL template we are working with
+   * @param {object} values - A key-value object containing the value for variables in the template.
+   * @returns {object} - A resulting gql object
+   */
+  const generateQueryFromTemplate = (template, values) => {};
+
+  /**
+   * Adds or removes field name from the selectedFields array.
+   * @param {string} fieldName - The name of the field.
+   */
+  const toggleField = fieldName => {
+    let newFieldList = selectedFields;
+
+    // If it is there, remove it.
+    if (isFieldEnabled(fieldName)) {
+      const index = newFieldList.indexOf(fieldName);
+      if (index !== -1) newFieldList.splice(index, 1);
+
+      // If it isn't there, then add it.
+    } else {
+      newFieldList.push(fieldName);
+    }
+    const newList = [...newFieldList];
+    setSelectedFields(newList);
+  };
+
+  /**
+   * Toggles the diff only option in state
+   */
+  const toggleDiffOnly = () => {
+    setShowFieldsDiffOnly(!showFieldsDiffOnly);
+  };
+
+  /**
+   * Returns the existing record in database.
+   * @returns {object|null} - The object data from the database
+   */
+  const getOriginalRecord = () => {
+    return recordData["atd_txdot_crashes"][0] || null;
+  };
+
+  /**
+   * Returns the new record as an object
+   * @returns {object|null} - The parsed object
+   */
+  const getNewRecord = () => {
+    try {
+      return JSON.parse(recordData["atd_txdot_changes"][0]["record_json"]);
+    } catch {
+      redirectToQueueIndex();
+      return null;
+    }
+  };
+
+  /**
```

Comment:
I need to get in the habit of creating more helpers to make life easier like these. 👍